plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+            same instance.
+        """
+
+        try:
+            payload = {"id": vn_id}
+            task_name = "delete_layer3_virtual_network_by_id"
+            task_id = self.get_taskid_post_api_call("sda", task_name, payload)
+
+            if not task_id:
+                self.msg = "Unable to retrive the task_id for the task '{0}'.".format(task_name)
+                self.set_operation_result("failed", False, self.msg, "ERROR")
+                return self
+
+            success_msg = "Layer3 Virtual Network '{0}' deleted successfully from the Cisco Catalyst Center.".format(vn_name)
+            self.get_task_status_from_tasks_by_id(task_id, task_name, success_msg)
+            self.deleted_vn.append(vn_name)
+
+        except Exception as e:
+            self.msg = "Exception occurred while deleting the layer3 Virtual Network '{0}' due to: {1}".format(vn_name, str(e))
+            self.set_operation_result("failed", False, self.msg, "ERROR").check_return_status()
+
+        return self
+
+    def is_ip_pool_exist(self, ip_pool_name, site_id):
+        """
+        Checks if a specified IP pool exists in the Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            ip_pool_name (str): The name of the IP pool to check for existence.
+            site_id (str): The identifier of the site where the IP pool is located.
+        Returns:
+            bool: True if the IP pool exists, False otherwise.
+        Description:
+            This function sends a request to the Cisco Catalyst Center to retrieve information
+            about a specific reserved IP subpool based on the provided IP pool name and site ID.
+            The primary purpose of this function is to facilitate validation of IP pool existence
+            for network configurations or management tasks.
+        """
+
+        try:
+            response = self.dnac._exec(
+                family="network_settings",
+                function="get_reserve_ip_subpool",
+                op_modifies=True,
+                params={
+                    "site_id": site_id,
+                    "group_name": ip_pool_name
+                }
+            )
+            response = response.get("response")
+            self.log("Received API response from 'get_reserve_ip_subpool' for the IP Pool '{0}': {1}".format(ip_pool_name, str(response)), "DEBUG")
+
+            if not response:
+                self.log("There is no reserve ip pool '{0}' present in the Cisco Catalyst Center system.".format(ip_pool_name), "INFO")
+                return False
+
+        except Exception as e:
+            self.msg = (
+                "Error while getting the details for reserve IP Pool with name '{0}' present in "
+                "Cisco Catalyst Center: {1}"
+            ).format(ip_pool_name, str(e))
+            self.set_operation_result("failed", False, self.msg, "ERROR").check_return_status()
+
```

Comment:
self.log("IP Pool '{0}' exists in the Cisco Catalyst Center.".format(ip_pool_name), "INFO")